### PR TITLE
Releasing v0.0.2 new functions and fixing some important bugs

### PR DESCRIPTION
**Pull request by SHIMA0111**
### Summary
Release v0.0.2 of `templatia` and `templatia-derive`, introducing new option-handling features, safer template validation, clearer defaults, and important API cleanups. This PR updates documentation and tests accordingly.

### What’s new (Added)
- Support for `Option<T>`: when a placeholder is absent in the template, fields of type `Option<T>` automatically default to `None`.
- New attribute `#[templatia(empty_str_option_not_none)]`: treat empty strings for `Option<String>` as `Some("")` instead of the default `None`.
- New attribute `#[templatia(allow_missing_placeholders)]`: allow fields not present in the template; such fields are initialized with `Default::default()` (requires `Default` for those field types).
- Compile-time validation to prevent ambiguous consecutive placeholders (e.g., consecutive `String`-like fields) with clearer diagnostics.
- Tests: comprehensive round-trip tests, option-handling tests, manual implementation examples, and compile-fail cases.
- Documentation: `LIMITATION.md` (current constraints) and `ROADMAP_CHANGES.md` (short-term roadmap).
- Documented MSRV: Rust 1.85.0 (Edition 2024).

### Changed
- README updates (EN/JA) to reflect new attributes, defaults, and examples, including explicit description of default template generation when no template is provided.
- Internal derive generator refactoring plus utilities/validator updates to support new parsing and validation logic.
- `AGENTS.md` updated to clarify testing and documentation conventions.

### Fixed
- More consistent placeholder handling across edge cases in parsing.
- Adjusted compile-fail tests for missing fields to reflect new validation behavior.

### Breaking changes
- `Template` trait API update:
  - Removed associated type `type Struct`.
  - `from_str` now returns `Self` (was `Self::Struct`).
- Method renames:
  - `from_string` -> `from_str`
  - `to_string` -> `render_string`
- The `#[derive(Template)]` macro has been updated to generate the new signatures and method names. Manual implementations must update accordingly.

### Rationale
- Clarify default behaviors for optional and missing data to reduce boilerplate and surprising runtime errors.
- Improve compile-time safety for ambiguous templates.
- Align naming with common Rust conventions (`from_str`, `render_string`).

### MSRV
- Rust 1.85.0 (Edition 2024). Documented

### Docs and examples
- Updated `README.md` / `README-ja.md` with examples for `Option<T>`, `allow_missing_placeholders`, and `empty_str_option_not_none`.
- Added limitations and roadmap docs to set expectations and guide users.

### Testing
- Added round-trip and option-behavior tests, manual implementations, and compile-fail test coverage.
- Ensures backward-compat with derived code after applying the breaking API renames.

### Migration guide (short)
- Update manual impls of `Template`:
  - Remove `type Struct`.
  - Change `fn from_str(s: &str) -> Result<Self, Self::Error>`.
  - Rename `to_string` -> `render_string` and `from_string` -> `from_str`.
- Rebuild projects using `#[derive(Template)]` to pick up new method names.

### Checklist
- [x] Changelog updated for `0.0.2`
- [x] Docs updated (EN/JA)
- [x] Tests added/updated
- [x] Breaking changes documented

### Impact
- Users relying on the old `Template` API must update method names/signatures.
- Template behavior is safer by default; optional and missing data are handled more predictably.

### Notes
- Title: Releasing v0.0.2 new functions and fixing some important bugs
- Target branch: `main`　
- Date: 2025-10-03

If you need additional migration details or release notes formatting for crates.io, I can provide a condensed version as well.

## Summary by CodeRabbit

- New Features
  - Added optional placeholder handling with attributes: allow_missing_placeholders and empty_str_option_not_none.
  - Improved Option<T> behavior, including empty-string handling and clearer parsing for optional booleans.

- Refactor
  - Breaking API: to_string/from_string renamed to render_string/from_str; parsing now returns Self (associated type removed).

- Documentation
  - Updated README (EN/JA) and CHANGELOG to new API; added LIMITATION.md; clarified roadmap and contributor guidance.

- Tests
  - Extensive new coverage for Option<T>, missing placeholders, round-trips, and edge cases.

- Chores
  - Updated crate keywords and categories.

### CODE REVIEW

File: templatia-derive/src/utils.rs [hunk 1]
```diff
@@ -33,4 +33,30 @@ pub(crate) fn get_type_name(ty: &syn::Type) -> String {
...
+                        if let Some(syn::GenericArgument::Type(inner_type)) = args.args.first() {
+                            return Some(inner_type);
+                        }
+                    }
+                }
+            }
+        }
+    }
+    None
 }
```
[the remark this answers is not in the document]
Comment (SHIMA0111) [reply templatia-derive/src/utils.rs:62]: @coderabbitai
This is pub(crate) so currently I decide to not describe the rustdoc. This is not a public but crate internal public.
